### ISSUE TITLE
Stop calling ElasticsearchMappings#addDocMappingIfMissing from transport thread

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#87911]
In #87679 we forbid calls to `Metadata#findMappings` on the latency-sensitive transport threads because it can be rather expensive. This method is called from `ElasticsearchMappings#addDocMappingIfMissing` which has 17 uses, all in ML-related code, and many of which seem to have no protection against happening on a transport thread. We work around this in that PR by adding some code that makes a decision to fork based on the thread name within the method itself. This is kind of inelegant, it would be better for the callers to choose the right thread in the first place, so I'm opening this issue to track this piece of technical debt.

Relates https://github.com/elastic/elasticsearch/issues/86765 since most of the callers end up on transport threads because they're ultimately called in a response handler for a client action.

**Comment by MananJethwani** [comment]
@DaveCTurner I would like to contribute to this issue, is this still open?

**Comment by DaveCTurner** [comment]
Thanks @MananJethwani, the issue is still open indeed. It might be a little tricky for a first contribution since it needs a reasonable level of understanding of Elasticsearch's threading model, but I'll let @droberts195 offer further guidance here since he added the `good first issue` label.

**Comment by droberts195** [comment]
Yes, with hindsight maybe this isn’t a good first issue after all. I’ve removed the label.

**Comment by valeriy42** [comment]
Severity: Could
Rationale: Internal threading/architecture debt, not a user-visible defect. A workaround already forks off transport threads inside ElasticsearchMappings#addDocMappingIfMissing (per [#87679](https://github.com/elastic/elasticsearch/pull/87679)). ML keeps working; the issue is inelegance and latent latency risk on transport threads.

Urgency: Low
Rationale: Open since June 2022 with no escalation. No active outage or release blocker. Needs deep threading-model knowledge; good first issue was removed.

Context
Affected users: ML operators indirectly (possible transport-thread latency under load)
Workaround: Yes — in-method thread fork based on thread name
Business impact: Performance/stability hygiene, not feature breakage
Technical risk: Transport-thread blocking if the workaround regresses; harder maintenance across 17 ML call sites
